pydantic/validators.py

```diff
@@ -80,17 +80,20 @@ def bytes_validator(v: Any) -> bytes:
         raise errors.BytesError()
 
 
-BOOL_STRINGS = {'1', 'TRUE', 'ON', 'YES'}
+BOOL_TRUE = {True, 1, 'y', 'Y', 'yes', 'Yes', 'YES', 'true', 'True', 'TRUE', 'on', 'On', 'ON', '1', 't', 'T'}
+BOOL_FALSE = {False, 0, 'n', 'N', 'no', 'No', 'NO', 'false', 'False', 'FALSE', 'off', 'Off', 'OFF', '0', 'f', 'F'}
 
 
 def bool_validator(v: Any) -> bool:
     if isinstance(v, bool):
```

Comment:
I kept this logic fork (even though redundant) so you can retain maximum performance by using actual bools.